module/documents/chat-message.mjs

```diff
@@ -166,26 +166,36 @@ export default class ChatMessage5e extends ChatMessage {
         }
       });
 
-      // If the user is the message author or the actor owner, proceed
       let actor = game.actors.get(this.speaker.actor);
-      if ( game.user.isGM || actor?.isOwner || (this.author.id === game.user.id) ) {
-        const optionallyHide = (selector, hide) => {
-          const element = chatCard[0].querySelector(selector);
-          if ( element && hide ) element.style.display = "none";
-        };
-        optionallyHide('button[data-action="summon"]', !SummonsData.canSummon);
-        optionallyHide('button[data-action="placeTemplate"]', !game.user.can("TEMPLATE_CREATE"));
-        optionallyHide('button[data-action="consumeUsage"]', this.getFlag("dnd5e", "use.consumedUsage"));
-        optionallyHide('button[data-action="consumeResource"]', this.getFlag("dnd5e", "use.consumedResource"));
-        return;
+      const isCreator = actor?.isOwner || (this.author.id === game.user.id);
```

Comment:
If a message is visible to the creator only, it should probably be visible to the DM too.